third_party/musl/crt/patch.diff

```diff
@@ -497,7 +497,30 @@ index afec985a..acf447c4 100644
  	return p;
  }
  
-@@ -2302,7 +2356,7 @@ static void error(const char *fmt, ...)
+@@ -2272,6 +2326,9 @@ no_redir:
+ 	return __dlsym(p, s, ra);
+ }
+ 
++/* Lock to keep calls to callbacks thread-safe. 
++ * Passed callbacks may have static members. */
++static pthread_mutex_t dl_iterate_cb_lock;
+ int dl_iterate_phdr(int(*callback)(struct dl_phdr_info *info, size_t size, void *data), void *data)
+ {
+ 	struct dso *current;
+@@ -2287,9 +2344,11 @@ int dl_iterate_phdr(int(*callback)(struct dl_phdr_info *info, size_t size, void
+ 		info.dlpi_tls_modid = current->tls_id;
+ 		info.dlpi_tls_data  = current->tls.image;
+ 
++		pthread_mutex_lock(&dl_iterate_cb_lock);
```

Comment:
It is possible to call dl_iterate_phdr from within a callback. Using a mutex would lead to deadlocks.